packages/@uppy/aws-s3/src/index.js

```diff
@@ -55,6 +55,12 @@ module.exports = class AwsS3 extends Plugin {
 
     this.opts = { ...defaultOptions, ...opts }
 
+    if (!this.opts.includes('companionUrl')) {
```

Comment:
We should only warn if there is no user-provided getUploadParameters function:
```suggestion
    if (!this.opts.companionUrl && !opts.getUploadParameters) {
```

(and check for the `companionUrl` _property_ because it's an object rather than an array)

Reply:
Oh I overlooked that 😳